transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -283,12 +284,13 @@ protected void run() {
                         break;
 
                     case SelectStrategy.SELECT:
-                        if (wakenUp == 1) {
-                            wakenUp = 0;
-                        }
+                        wakenUp = 0;
                         if (!hasTasks()) {
                             strategy = epollWait();
                         }
+                        if (strategy > 0 || wakenUp == 0) {
+                            wakenUp = 1;
```

Comment:
I think wakenUp might not be the best name for this anymore.  What do you think about `disableWakeUp` ?   If 0, other threads may write to the event FD.  If 1, the event loop will handle any tasks before calling epollWait (but not epollWaitNow).

Reply:
How about just `awake`?

Reply:
The more I see how this logic is changing, the more it seems to be more similar to https://www.scylladb.com/2018/02/15/memory-barriers-seastar-linux/ on the section `The Complications of Sleeping` :)

Reply:
@franz1981 interesting, that does look similar, and maybe even more so to the the next iteration that I was playing with to remove the debouncing CAS/GAS completely and just piggyback on the queue's coordination: fa62b0ad1e7bc7d4b01266c6f45bb7466f3db029

It is in a hacked but fully-working state, let me know what you think :) though we would have to request the new "offerAndReturnProducerIndex" jctools mpsc queue method unless there's a way around it that i haven't thought of...

Reply:
cc @nitsanw